src/odfesqlodbc/es_statement.cpp

```diff
@@ -148,6 +151,36 @@ RETCODE ExecuteStatement(StatementClass *stmt, BOOL commit) {
     return CleanUp();
 }
 
+SQLRETURN GetNextResultSet(StatementClass *stmt) {
+    if (stmt == NULL) {
+        return SQL_ERROR;
+    }
+    ConnectionClass *conn = SC_get_conn(stmt);
+    QResultClass *q_res = SC_get_Result(stmt);
+    if ((q_res == NULL) && (conn == NULL)) {
+        return SQL_ERROR;
+    }
+    SQLSMALLINT total_columns = -1;
+    SQLNumResultCols(stmt, &total_columns);
+    if (total_columns == -1) {
```

Comment:
Is 0 columns acceptable? Should this not check the return value of SQLNumResultCols?

Reply:
0 is used for a bookmark. SQLNumResultCols takes a pointer to a buffer to return the number of columns in the result set. So here I am checking if the buffer is updated, return an error if it isn't updated.

Reply:
Shouldn't the return of the function also be checked then?

Reply:
Yes, I will add